File: src/main/java/iot/tetracubered/configurations/MessagingClientConfiguration.java
package iot.tetracubered.configurations;

import org.eclipse.microprofile.config.inject.ConfigProperty;

public interface MessagingClientConfiguration {

    @ConfigProperty(name = "host")
    String host();

    @ConfigProperty(name = "port")
    Integer port();

    @ConfigProperty(name = "client-id")
    String clientId();

    @ConfigProperty(name = "user-name")
    String userName();

    @ConfigProperty(name = "password")
    String password();

    @ConfigProperty(name = "topics")
    TopicsConfiguration topics();
}

File: src/main/kotlin/iot/tetracubered/configurations/MessagingClientConfiguration.kt
package iot.tetracubered.configurations

import org.eclipse.microprofile.config.inject.ConfigProperty

interface MessagingClientConfiguration {

    @ConfigProperty(name = "host")
    fun host(): String

    @ConfigProperty(name = "port")
    fun port(): Int

    @ConfigProperty(name = "client-id")
    fun clientId(): String

    @ConfigProperty(name = "user-name")
    fun userName(): String

    @ConfigProperty(name = "password")
    fun password(): String

    @ConfigProperty(name = "topics")
    fun topics(): TopicsConfiguration
}